File: question-service/src/main/java/com/aya/questionservice/model/dto/QuestionOutputDto.java
package com.aya.questionservice.model.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;

@Data
@Builder
@AllArgsConstructor
public class QuestionOutputDto {

    private Integer id;
    private String questionTitle;
    private String option1;
    private String option2;
    private String option3;
    private String option4;
}

File: quiz_application/lib/models/QuestionOutputDto.dart
// models/question_output_dto.dart
class QuestionOutputDto {
  final int id;
  final String questionTitle;
  final String option1;
  final String option2;
  final String option3;
  final String option4;

  QuestionOutputDto({
    required this.id,
    required this.questionTitle,
    required this.option1,
    required this.option2,
    required this.option3,
    required this.option4,
  });

  factory QuestionOutputDto.fromJson(Map<String, dynamic> json) {
    return QuestionOutputDto(
      id: json['id'],
      questionTitle: json['questionTitle'],
      option1: json['option1'],
      option2: json['option2'],
      option3: json['option3'],
      option4: json['option4'],
    );
  }
}
